Dockerfile-master

```diff
@@ -1,24 +1,17 @@
 FROM node:8-alpine
 
-ARG VERSION
-ARG TYPE=basic
-
 ENV NODE_ENV production
-# this will enable polling, hot-reload will work on docker or network volumes
+
+# Enable chokidar polling so hot-reload mechanism can work on docker or network volumes
 ENV CHOKIDAR_USEPOLLING true
 
 RUN mkdir -p /usr/src/app
 WORKDIR /usr/src/app
 
-COPY package.json /usr/src/app/
+COPY package.json package-lock.json /usr/src/app/
 RUN npm install
 COPY . /usr/src/app
 
-EXPOSE 8080
-
-# HTTPS
-# EXPOSE 8443
+EXPOSE 8080 9876
 
-# Admin API
-EXPOSE 9876
-CMD [ "npm", "run", "start" ]
+CMD [ "node", "lib", "index.js" ]
```

Comment:
Before you ask "Why not use `npm start`?" — it's because npm start, in docker environments, is swallowing signals (such as `SIGKILL`, `SIGTERM`) and not sending them to `node`, that should handle them accordingly.